[1.2] Move Command / Delete Command

lormee94 opened the pull request:
As per #102 #30 
- "move" command now usable (e.g move 1 C:\Users\ [user]\Desktop)
- "edit" command cannot make changes to directory

As per #49 
- added option for hard delete, with keyword "hard" added behind index
- keyword is not case sensitive

#LATB

lormee94 committed:
(v1.2) Move Command
- Created move command and removed the need for directory prefix.
- Create move command parser

lormee94 committed:
(v1.2) Move Command
- Nerf edit command to not accept changes in directory
- Removed some code from Move and Edit command that is redundant

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main

lormee94 committed:
[v1.2] Delete Command
- Added option for hard delete, with keyword "hard" added behind index
- keyword is not case sensitive

ptvrajsk commented on src/main/java/seedu/address/logic/commands/DeleteCommand.java:
```diff
@@ -11,24 +11,37 @@
             + ": Deletes the PDF identified by the index number used in the displayed PDF list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + "Hard or soft delete can be chosen (soft delete selected by default).\n"
```
I think we can remove the soft delete can be chosen part? Cause its by default.

ptvrajsk replied:
Also I think must specify what hard delete does? Or do we just do it in the UG & DG?

ptvrajsk commented on src/main/java/seedu/address/logic/commands/DeleteCommand.java:
```diff
@@ -11,24 +11,37 @@
+            + "Hard or soft delete can be chosen (soft delete selected by default).\n"
+            + "Parameters(Soft Delete): INDEX (must be a positive integer)\n"
+            + "Parameters(Hard Delete): INDEX (must be a positive integer) hard\n"
+            + "Example: " + COMMAND_WORD + " 1 | " + COMMAND_WORD + " 1 hard";
```
And I think maybe can remove the " | " cause like it might be confusing for a non-technical user. We could just put "OR" instead?

ptvrajsk commented on src/main/java/seedu/address/logic/commands/MoveCommand.java:
```diff
@@ -0,0 +1,106 @@
+            + "Example: " + COMMAND_WORD + " 1 Desktop\\main\\target";
+
+    public static final String MESSAGE_MOVE_PDF_SUCCESS = "Moved PDF: %1$s";
+    public static final String MESSAGE_NOT_MOVED_SAME = "Why are you moving the PDF to the same place?";
```
HAHA Nice la

ptvrajsk replied:
👍

ptvrajsk commented on src/main/java/seedu/address/logic/commands/MoveCommand.java:
```diff
@@ -0,0 +1,106 @@
+            + "by the index number used in the displayed pdf list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[NEW_DIRECTORY]\n"
+            + "Example: " + COMMAND_WORD + " 1 Desktop\\main\\target";
```
Should we put an actual location like "C:\Users\....\main..\target" instead of  "Desktop\\main\\target"

ptvrajsk commented on src/main/java/seedu/address/logic/commands/DeleteCommand.java:
```diff
@@ -11,24 +11,37 @@
 import seedu.address.model.Model;
 import seedu.address.model.pdf.Pdf;
 
+import java.io.File;
```
The Import statements are in the wrong order bro. I think can try to resolve some of the checkstyle errors mentioned by Travis and push again? Will save us sometime towards the end to get the builds passing haha.

lormee94 committed:
[v1.2] Fixes to Travis Build fail
- enough said.

lormee94 commented:
Thanks for all the feedback, working on rectifying them.

lormee94 committed:
[v1.2] Fixing Travis Build Fail

lormee94 committed:
samasama

lormee94 committed:
[v1.2] Fixing Travis Build fail (test portion)
- Some refactoring of persons to pdf
- Modify the Delete command constructor so it has normal function and accepted by tests.

lormee94 committed:
[v1.2] Fixing Travis Build fail
- Modifications to DeleteCommand file

lormee94 committed:
[v1.2] Fixing Travis Build Fail
- Made edits to UserGuide to resolve tagging issue that was seen in build fail
- LATB

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main

lormee94 committed:
[v1.2] Address merge conflicts from recent PR#114

lormee94 committed:
samasama

lormee94 committed:
samasama

lormee94 committed:
samasama

lormee94 committed:
samasama

lormee94 committed:
samasama

ptvrajsk commented:
Great!
